cache/lib/cache/s3.ex

```diff
@@ -53,7 +53,32 @@ defmodule Cache.S3 do
   Returns :ok on success, {:error, :rate_limited} on 429 errors (should be retried),
   or {:error, reason} on other failures.
   If the local file does not exist, returns :ok (the file may have been evicted).
+
+  Supports both CAS artifacts and module cache artifacts. Module cache artifacts
+  are identified by an artifact_id starting with "module::".
   """
+  def upload(account_handle, project_handle, "module::" <> rest) do
+    [category, hash, name] = String.split(rest, "::", parts: 3)
+    key = Cache.Disk.module_key(account_handle, project_handle, category, hash, name)
```

Comment:
should we consider to flatten the cache artifacts instead? I find this prefixing with the category a bit odd. Or if we need to distinguish between different types of cache, why not add dedicated endpoints for the module cache?